Initial import (#1)
mdn/temp-processes: pull-requests.md

4. **Publishing:** Updated content goes live within a day of merging via a site rebuild once every 24 hours.

### Decision tree

TODO: example decision tree in mermaid

```mermaid
graph TB
  A("Do you think<br> online learning is right for you?")
  B("Do you have time to design<br> a service learning component?")
  C("What is the civic or public purpose of your discipline?<br> How do you teach that without service learning?")
  D("Do you have departmental or school<br> support to plan and implement service learning?")
  E["Are you willing to be a trailblazer?"]
  F["What type of service learning to you want to plan?"]

  A==Yes==>B
  A--No-->C
  B==Yes==>D
  B--No-->C
  D--Yes-->F
  D--No-->E
  E--Yes-->F
  E--No-->C
```

## Guidelines for opening a pull request

### Before you open a pull request

Anyone who plans to open a PR to make a change on MDN Web Docs should:

- **Open a discussion** to describe why you think the changes are required to improve the content.
  This is important if your PR is a technical change, not typo fixes, grammatical improvements, formatting or structural changes.
- **File an issue describing what the PR will change.**

>   When you submit a PR, reference this issue using `#` followed by the issue's ID, for example `#1234`.

Reviewer: I think we can split this into:

* Opening a PR because you want to make a change
* Opening a PR to fix an existing issue (for instance from the contributors project board)

We should add the content proposal process here. Also folks are directed to raising a new issue from MDN directly - ok ok I guess this is a case for a flow chart :)

Author: 😄 I think this is more clear here after https://github.com/mdn/temp-processes/pull/1/commits/ea80118c479ccb45cb3d20a5adac49d143ce6bdf

### Opening a pull request

- **PRs should be short and focused to one issue** or a related set of changes.
  If a PR becomes too large, the reviewer may close your PR and ask that you submit a separate PR for each logical set of changes that belong together.
- **PRs should not contain large amounts of grammar updates**.
  MDN Web Docs contains technical documentation; you should not report improvements in the grammar, but cases where the grammar is incorrect are acceptable.
- **PRs that update API usage need an accompanying mdn/content PR** to update the corresponding documentation.

>   PRs of this type may be rejected if there is no corresponding content PR.

Reviewer: I think this is suppose to be concerning if demo (interactive examples, dom examples, learn area et al) updates are needed.

If demo updates happen is there a corresponding content pr
If content and in page code examples are updated, are the corresponding demos updated

- **Don't enable auto-merge.**
- **Add 'depends on'** if there are PRs that must land first (e.g. linked examples in other repos)

### After you open a pull request

- **Handle CI failures** from the automated tests run as GitHub Actions (see `.github/workflows`).
  If one or more of these tests fail, it is your responsibility to try and resolve them.
  Your PR will not be approved and merged if these tests are failing.
  If you don't know how to resolve the underlying issues, ask for help.

- **Resolve merge conflicts** with the main branch; you are responsible for resolving these.
  You can do this by merging the main branch into your branch (`git pull mdn main`), and pushing the updated branch to your fork (git push).

> - **Don't reopen closed PRs.** If you must create a new PR, it can reference the closed one.

Reviewer: + Exercise patience with the review process, the mdn org receives many pull requests a day so please allow reviewers time to get to reviewing your pr.

+ Be prepared to make changes to the pull request based on the review. If a review happens and the changes are not made, the pr may be closed

Author: Good, have updated

  - If you are not sure how to make changes to someone else's PR, @vkWeb wrote some nice simple instructions on how to do this from the command line; see ReviewPRCommands.
  - Alternatively, you can edit files using the GitHub UI — go to the pull request's "Files changed" tab, find the file you want to edit, and choose "three dot" menu (...) > Edit file.
- Merging PRs: The assigned PR reviewer merges the PR. This can also be the reviewer you asked for help with the review.
- Resolving merge conflicts: You may encounter merge conflicts as you review PRs if another PR that touches some of the same files got merged before the one you are reviewing. Addressing merge conflicts on GitHub is a useful resource to help you. Feel free also to ask your team(s) for help if you need it.

### If you are assigned to review a PR

- **Code examples should follow guidelines.**

> - **Check the MDN Writing style guide**, and use it to inform your review new content.

Reviewer: ```suggestion
- **Check the MDN Writing style guide**, and use it to inform your review of new content.
```

# Pull Request submission and review guidelines

## Purpose of this document

The objective of this document is to have a single place with information from the following pages:

- **PR etiquette:** https://github.com/mdn/content#pull-request-etiquette
- **Reviewing PRs:** https://github.com/mdn/content/blob/main/REVIEWING.md
- **Open source etiquette:** https://developer.mozilla.org/en-US/docs/MDN/Contribute/Open_source_etiquette

## Contributing to MDN

This section describes how contributors make changes on MDN Web Docs and how the changes are reviewed and land on the site.
Content changes to MDN Web Docs are related to a variety of work streams, including:

- **Day-to-day content improvements:** documentation of new APIs, CSS properties, platform updates and content additions.
  This is usually done by MDN Web Docs staff working for Mozilla, Google, Open Web Docs, Samsung, but also by community volunteers.
- **Minor fixes:** small updates done to the site to fix typos, grammatical issues, and technical inaccuracies.
  These issues are usually found by users of MDN Web Docs.
- **Content bug fixes:** usually done by volunteers to close issues on `mdn/content` repo.

> Regardless of how content changes are done, they are submitted as pull requests (PRs) on GitHub:

Reviewer 2: How about we refer to them as stages - submission, review, approval

```suggestion
Regardless of how content changes are done, they are submitted as pull requests on GitHub. The content changes go through the following stages before they are published on MDN Web Docs:

1. **Submission:** As a pull request author, you submit the changes. There are a few guidelines you need to be aware of before and after you open a pull request.
```
This would need to be linked to the guideline section below.

> 1. **Reviewers** comprised of MDN staff members and volunteers are "topic review owners".

Reviewer 2: ```suggestion
2. **Review:** Your changes are reviewed by MDN staff members and volunteers.
```

   When a pull request related to a particular topic is opened, it will be assigned to that area's topic review owners.
   The `CODEOWNERS` file specifies the teams who own certain content directories.

> 2. **Approved pull requests** can be merged after a approving review has been made.

Reviewer 2: ```suggestion
2. **Approval:** The changes can be merged after a approving review has been made.
```

   The assigned reviewer will merge the pull request.

> 3. **Content update** is done automatically via a site rebuild once every 24 hours.
>    Volunteers can see their changes go live within a day of merging.

Reviewer 2: What would be a good name for this stage that aligns with 'Submission' > 'Review' > 'Approval'?
Or perhaps we don't call this a stage..

```suggestion
Updated content goes live within a day of merging via a site rebuild once every 24 hours.
```

Author: "Publishing" maybe

### Decision tree

TODO: example decision tree in mermaid

```mermaid
graph TB
  A("Do you think<br> online learning is right for you?")
  B("Do you have time to design<br> a service learning component?")
  C("What is the civic or public purpose of your discipline?<br> How do you teach that without service learning?")
  D("Do you have departmental or school<br> support to plan and implement service learning?")
  E["Are you willing to be a trailblazer?"]
  F["What type of service learning to you want to plan?"]

  A==Yes==>B
  A--No-->C
  B==Yes==>D
  B--No-->C
  D--Yes-->F
  D--No-->E
  E--Yes-->F
  E--No-->C
```

> 

Reviewer 2: ```suggestion
## Guidelines for opening a pull request
```

### Before you open a pull request

Anyone who plans to open a PR to make a change on MDN Web Docs should:

- **Open a discussion** to describe why you think the changes are required to improve the content.
  This is important if your PR is a technical change, not typo fixes, grammatical improvements, formatting or structural changes.
- **File an issue describing what the PR will change.**
  When you submit a PR, reference this issue using `#` followed by the issue's ID, for example `#1234`.

> ### Making a pull request

Reviewer 2: ```suggestion
### Opening a pull request
```

- **PRs should be short and focused to one issue** or a related set of changes.
  If a PR becomes too large, the reviewer may close your PR and ask that you submit a separate PR for each logical set of changes that belong together.
- **PRs should not contain large amounts of grammar updates**.
  MDN Web Docs contains technical documentation; you should not report improvements in the grammar, but cases where the grammar is incorrect are acceptable.
- **PRs that update API usage need an accompanying mdn/content PR** to update the corresponding documentation.
  PRs of this type may be rejected if there is no corresponding content PR.
- **Don't enable auto-merge.**
- **Add 'depends on'** if there are PRs that must land first (e.g. linked examples in other repos)

> ### After you have opened a pull request

Reviewer 2: ```suggestion
### After you open a pull request
```

- **Handle CI failures** from the automated tests run as GitHub Actions (see `.github/workflows`).
  If one or more of these tests fail, it is your responsibility to try and resolve them.
  Your PR will not be approved and merged if these tests are failing.
  If you don't know how to resolve the underlying issues, ask for help.

- **Resolve merge conflicts** with the main branch; you are responsible for resolving these.
  You can do this by merging the main branch into your branch (`git pull mdn main`), and pushing the updated branch to your fork (git push).

- **Don't reopen closed PRs.** If you must create a new PR, it can reference the closed one.

> ## Pull request assignees

Reviewer 2: ```suggestion
## Guidelines for reviewing a pull request
```

reviewer: @jpmedley
Upon submitting the PR, they will request a review from the reviewer specified in the PR description. Once that reviewer has approved the new content, they will then ask you for an approval as required by the CODEOWNERS system for the pull request to be mergeable.
Therefore, if you receive a PR review request and then see that you have been overridden with another reviewer in the manner described above, then don't review the changes — just wait for an approval request.
Guidelines for reviewing a pull request
These guidelines apply to anyone who is tasked with reviewing MDN Web Docs content PRs.
Prerequisite knowledge and Github tools

The following specific topic areas are being reviewed by the kind souls listed underneath them. Be kind to them, and thank them for all the help they give to this project. If you would like to help with MDN content reviews, get in touch with us.
Note that changes to any content areas not explicitly listed below will be handled by the @core-yari-content team, which currently consists of @Rumyra.

> - Web accessibility content:

Author: A table might be nice for this. It needs to be formatted differently, the list is too long / hard to read in this state

Reviewer 2: +1 for table. Also, this information is pretty outdated. Up for discussion - maybe we should publish only the team aliases

Author: Shall we remove it for now?
